src/controllers/EventParticipantsController.ts

```diff
@@ -0,0 +1,39 @@
+import { JoinedPeople } from '../models'
+import { Request, Response } from 'express'
+import { Message } from '../config/messages'
+import CustomError from '../helpers/CustomError'
+import validateParams from '../validation/paramsId'
+export default class EventParticipantsController {
+  public static async index (req: Request, res:Response):Promise<void> {
+    const { eventId } = req.params
+    await validateParams({ id: eventId })
+    const allJoined = await JoinedPeople.findAll({
+      where: { EventId: eventId }
+    })
+    if (!allJoined) throw new CustomError(Message.NOT_FOUND, 404)
+    res.json({ data: allJoined, message: Message.SUCCESS })
+  }
+
+  public static async store (req: Request, res:Response) {
+    const { eventId } = req.params
+    const { UserId } = req.body
+    await validateParams({ id: eventId })
```

Comment:
I believe you still need to validate `UserId` and I guess you can use the same function `validateParams` for that  but then its name will be unsuitable as it is came from body not params.
And what I also recommend when you have 2 or more independent async function (ex. when you don't need the result of one as an input for the other) is to put them together in [Promise.All](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) so they can launch together then take less time to go the next line
```suggestion
    const { eventId } = req.params
    const { UserId } = req.body
    await Promise.all([
    validateParams({ id: eventId }),
    validateParams({ id: UserId })
])

```

Reply:
As the userId will be taken from the token(req.token), I don't think we need to validate it. Once the user is logged in, he will be able to join the event, otherwise he will not be authorized to do so..

We did handle the signup & login, but the PRs are hanged

Reply:
Are you saying that this `const { UserId } = req.body`  just temporary and will be change later? if yes for things similar to this situation please write a comment in your code which describes that.
Because I thought it An Admin controller so can control other users not just him/herself

Reply:
Yup, exactly